### PR TITLE
[DBT] Implement adapter.rename_relation

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Reference: https://docs.getdbt.com/reference/dbt-jinja-functions/adapter#rename_relation

### CODE REVIEW

**Comment by crericha** [comment]
In order for this functionality to work, I think we'd need to convert models to sqlmesh format in dag order, so the downstream models have the correct relation.

File: sqlmesh/dbt/adapter.py [hunk 1]
```diff
@@ -209,6 +216,14 @@ def get_missing_columns(
             if col.name not in target_columns
         ]
 
+    def rename_relation(self, from_relation: BaseRelation, to_relation: BaseRelation) -> None:
+        if from_relation.identifier is not None and to_relation.identifier is not None:
+            new_name = to_relation.identifier
```
Comment (crericha) [inline sqlmesh/dbt/adapter.py:221]: unnecessary variable

File: sqlmesh/dbt/adapter.py [hunk 2]
```diff
@@ -209,6 +216,14 @@ def get_missing_columns(
...
+            old_table = exp.table_(
+                from_relation.identifier, db=from_relation.schema, catalog=from_relation.database
+            )
+            self.engine_adapter.rename_table(old_table, new_name)
```
Comment (crericha) [inline sqlmesh/dbt/adapter.py:225]: Does this actually work? Will sqlmesh detect that it needs to recreate old_table?

File: sqlmesh/dbt/adapter.py (hunk 1 above)
Comment (crericha) [reply sqlmesh/dbt/adapter.py:221]: Can new_name be in a different schema or database? This implementation assumes its the same.

Comment (GeorgeSittas) [reply sqlmesh/dbt/adapter.py:221]: > Can new_name be in a different schema or database? This implementation assumes its the same.

Provided that this operation results in an `ALTER TABLE table RENAME TO table_name`, I assumed just the identifier would suffice.

Comment (GeorgeSittas) [reply sqlmesh/dbt/adapter.py]: > unnecessary variable

Removed.

**Pull request closed without merging**